lib/internal/test_runner/reporter/lcov.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const Transform = require('internal/streams/transform');
+
+// This reporter is based on the LCOV format, as described here:
+// https://ltp.sourceforge.net/coverage/lcov/geninfo.1.php
+// Excerpts from this documentation are included in the comments that make up
+// the _transform function below.
+class LcovReporter extends Transform {
```

Comment:
This can be written more cleanly with modern stream machinery but I'm 100% OK with landing and then refactoring.